File: cpp/Dynamic Programming/L_flipStrToMonotoneMin.cpp
#include<bits/stdc++.h>
using namespace std;

int minFlipsMonoIncr(string s, int counter_one  = 0, int counter_flip = 0) {
        for (auto ch : s) {
            if (ch == '1') {
                ++counter_one;
            } else {
                ++counter_flip;
            }
            counter_flip = std::min(counter_one, counter_flip);
        }
        return counter_flip;
    }

File: Python/Dynamic Programming/L_flipStrToMonotoneMin.py
def minFlipsMonoIncr(s: str) -> int:
        n = len(s)
        cnt0 = s.count('0')
        cnt1 = 0
        res = n - cnt0
        for i in range(n):
            if s[i] == '0': 
                cnt0 -= 1
            elif s[i] == '1':
                res = min(res, cnt1+cnt0)
                cnt1 += 1
        return res